File: MessagingAndroid/app/src/main/java/ma/tayeb/messaging_android/types/Customer.kt
package ma.tayeb.messaging_android.types

import java.util.UUID

data class Customer(
    val id: UUID,
    val fullName: String,
    val prospectId: String,
    val clientId: String,
    val conversation: Conversation
)

data class CustomerCreationRequest(
    val fullName: String,
    val clientId: String?,
    val prospectId: String?
)

File: messaging-frontend/src/types/customer.ts
import type { Conversation } from "./conversation"

export type Customer = {
    id: string,
    fullName: string,
    prospectId: string,
    clientId: string,
    conversation: Conversation
}

export type CustomerCreationRequest = {
    fullName: string,
    prospectId: string,
    clientId: string
}